Carbon.Cryptography/Functions/Unprotect-CString.ps1

```diff
@@ -201,10 +201,22 @@ function Unprotect-CString
                 $count = $certificates | Measure-Object | Select-Object -ExpandProperty 'Count'
                 if( $count -gt 1 )
                 {
-                    $msg = "Found $($count) certificates at ""$($PrivateKeyPath)"". Arbitrarily choosing the first " +
-                        'one. If you get errors, consider passing the exact path to the certificate you want to ' +
-                        'the "Unprotect-CString" function''s "PrivateKeyPath" parameter.'
-                    Write-Warning -Message $msg
+                    $certificates = $certificates | Where-Object { $_.HasPrivateKey -and $_.PrivateKey }
+                    $privateKeyCount = $certificates | Measure-Object | Select-Object -ExpandProperty 'Count'
+
+                    if( $privateKeyCount -gt 1 )
+                    {
+                        $msg = "Found $($privateKeyCount) certificates (which contain private keys) at ""$($PrivateKeyPath)"". " +
+                            'Arbitrarily choosing the first one. If you get errors, consider passing the exact path to ' +
+                            'the certificate you want to the "Unprotect-CString" function''s "PrivateKeyPath" parameter.'
+                        Write-Warning -Message $msg
+                    }
+                    elseif( $privateKeyCount -eq 0 )
+                    {
+                        "Found $($count) certificates at ""$($PrivateKeyPath)"" but none of them contain a private " +
+                        'key or the private key is null.' | Write-Error

```

Comment:
Add `This is usually because the certificate was installed without a private key or the current user doesn't have permission to read the private key.` to the end of the error message.

Reply:
Done.